### PR TITLE
Fixes strip_key and adds filters and proxy_mode

### PR DESCRIPTION
**Pull request by luisdavim**
These changes are to fix the `strip_key` function and to add a feature to rename metric names using `filters`.
This also implements a `proxy_mode`, inspired on the work by @terjesannum

### CODE REVIEW

**Comment by luisdavim** [comment]
Hi, @majormoses or @agoddard.
Can you take a look at this?

Thanks,
Luis

**Comment by luisdavim** [comment]
Thanks @majormoses I thought you would have access since you have access to the sensu-plugins org...

**Comment by mbbroberg** [comment]
Hey @luisdavim I'm just getting started as Community lead at Sensu and working on getting my arms around things. Work under the [`sensu-plugin` org](https://github.com/sensu-plugins/) is maintained by a small group of users. We're working on formal maintainers for extensions now as well. Discussion at: https://github.com/sensu-plugins/sensu-plugins-feature-requests/issues/21

**Comment by majormoses** [comment]
@luisdavim I have access now I will try to review this tonight and if not for sure some time this week.

**Comment by luisdavim** [comment]
Awsome, thanks @majormoses and @mbbroberg !

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,9 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 This CHANGELOG follows the format listed at [Keep A Changelog](http://keepachangelog.com/)
 
 ## [Unreleased]
+- Add proxy_mode
```
Comment (majormoses) [inline CHANGELOG.md]: These should be under an `### Added` header. Also the description says fixed strip_keys keys but this is not represented here. Please add it under a `### Fixed Header`

File: lib/sensu/extensions/influxdb/influx_relay.rb [hunk 2]
```diff
@@ -19,7 +19,38 @@ def flush_buffer
...
+            result = influxdb.post(post_data)
+            next if @influx_conf.key?(db) # this is to avoid the performance impact of checking the response everytime
+            result.callback do
+              if result.response =~ /.*error.*/
+                logger.error(result.response)
+                if result.response =~ /.*database not found.*/
+                  post_data = {}
+                  post_data[:body] = ''
+                  post_data[:query] = {
+                    'db' => db, 'precision' => p,
```
Comment (majormoses) [inline lib/sensu/extensions/influxdb/influx_relay.rb]: is there a reason  that `'precision' => p,` is not on its own line?

**Pull request merged**